### ISSUE TITLE
Array initialized with wrong values under heavy load

### ISSUE THREAD

**Issue by crbrz** [peachpiecompiler/peachpie#308]
```php
function test() {
	$values = array(
		'v1' => '',
		'v2' => ''
	);

	if(strlen($values['v1']) > 0) {
		throw new Exception('corrupted');
	}

	$values['v1'] = 'changed';
}

test();
```

If the code above is run from the default ASP.NET Core Empty project (`dotnet new web -lang PHP`) with a web stress tool that uses multiple threads as client then there will be failed requests (exception on line 8). On my PC (Intel i7 2600K CPU with 16GB RAM) it happens when using 20 threads.

In the generated .net code there is a static field `<arr>?0` that holds the array used to initialize `$values` and somehow the content of `$values` after it's changed ends up in it.

**Comment by jakubmisek** [comment]
You are right, this happens if this kind of 'constant' array is used for more than `0x7fff` times.

'array' implementation is going to be changed including this faulty mechanism. We'll try implement a quick fix for now tho.

**Comment by jakubmisek** [comment]
No it is a temporary partial fix. It just postpones the issue. The issue is still open.

**Comment by jakubmisek** [comment]
thanks! does https://github.com/peachpiecompiler/peachpie/pull/309 fix the issue for you?

**Comment by crbrz** [comment]
Yes, `$values` is correctly initialized no matter how many threads are hitting `test()` at the same time.
